src/components/permissions/rowLevelPolicyHelpers.ts

```diff
@@ -0,0 +1,327 @@
+import databaseConnect from '../databaseConnect'
+import { UserInfo, PermissionRow, RuleTypes } from './types'
+
+const SQLBuilder = require('json-sql-builder2')
+const jsonToSql = new SQLBuilder('PostgreSQL')
+
+// in { permissionPolicyId: 1, permissionNameId:2 }
+// out "pp1pn2"
+const getPermissionNameAbbreviation = ({ permissionPolicyId, permissionNameId }: PermissionRow) =>
+  `pp${permissionPolicyId}pn${permissionNameId}`
+
+// in { permissionPolicyId: 1, permissionNameId:2, templatePermissionId: 3 }
+// out "pp1pn2tp3"
+const getTemplatePermissionAbbreviation = (permissionRow: PermissionRow) =>
+  `${getPermissionNameAbbreviation(permissionRow)}tp${permissionRow.templatePermissionId}`
+
+// Returns new version of object with each key prefixed with prefix (and value converter to string)
+// in "prefixMe", { one: 1, two: 2}
+// out { prefixMe_one: "1", prefixMe_two: "2" }
+const remapObjectKeysWithPrefix = (prefix: string, object: Object) => {
+  const remappedObject: { [index: string]: string } = {}
+
+  Object.entries(object).forEach(([key, value]) => {
+    remappedObject[`${prefix}_${key}`] = String(value)
+  })
+
+  return remappedObject
+}
+
+/* Compiles JWT from userInfo and PerissionRows
+  in { userId: 1, ... }, [
+  {
+    templatePermissionRestrictions: {
+      restrictOne: 1,
+      restrictTwo: 2,
+    },
+    templateId: 1,
+    permissionPolicyId: 2,
+    permissionNameId: 3,
+    templatePermissionId: 4,
+  }]
+  out {
+    aud: 'postgraphile',
+    userId: 1,
+    pp2pn3: true,
+    pp2pn3tp4: true,
+    pp2pn3tp4_templateId: "1",
+    pp2pn3tp4_restrictOne: "1",
+    pp2pn3tp4_restrictTwo: "2"
+  }
+*/
+const compileJWT = (userInfo: UserInfo, templatePermissionRows: Array<PermissionRow>) => {
+  let JWT = { userId: userInfo.userId, aud: 'postgraphile' }
+
+  templatePermissionRows.forEach((permissionRow: PermissionRow) => {
+    const { templatePermissionRestrictions, templateId, templatePermissionId } = permissionRow
+
+    let permissionAbbreviation = getPermissionNameAbbreviation(permissionRow)
+
+    JWT = {
+      [permissionAbbreviation]: true,
+      ...JWT,
+    }
+    // It's possible to have permission name given to user without any templatePermissionIds
+    if (!templatePermissionId) return // 'continue' syntax for for each
+
+    permissionAbbreviation = getTemplatePermissionAbbreviation(permissionRow)
+    JWT = {
+      [permissionAbbreviation]: true,
+      ...remapObjectKeysWithPrefix(permissionAbbreviation, {
+        ...templatePermissionRestrictions,
+        templateId,
+      }),
+      ...JWT,
+    }
+  })
+  return JWT
+}
```

Comment:
Just looking at the returned JWT, the username is not present, just the userId. Don't we need the username as well?

Reply:
12b4c98e39d006c54e02522f8e6c5189497d459a

Also I made an issue to address this properly: https://github.com/openmsupply/application-manager-server/issues/178